source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -0,0 +1,277 @@
+      for (unsigned int i = 0; i <= depths_iso.size();)
+        if (std::fabs(latitude - latitudes_iso[i]) <= delta && std::fabs(longitude - longitudes_iso[i]) < delta)
+          {
+            return -depths_iso[i]*1000;
+          }
+        else if (std::fabs(latitude - latitudes_iso[i]) <= delta*2 && std::fabs(longitude - longitudes_iso[i]) < delta*2)
+          {
+            return -depths_iso[i]*1000;
+          }
+        else
+          i++;
+      Assert(false, ExcInternalError());
+      return 0;
+    }
+
+    /**
+     * Calculate distance from the center of the Earth to the surface of WGS84.
+     */
+
+    template <int dim>
+    double
+    AdiabaticBoundary<dim>::radius_WGS84(const double &theta) const
+    {
+      const double eccentricity    = 8.1819190842622e-2;
+      const double semi_major_axis = 6378137.0;
+      return semi_major_axis/std::sqrt(1- eccentricity * eccentricity
+                                       * std::sin(numbers::PI*theta/180)*std::sin(numbers::PI*theta/180));
+    }
+
+    template <int dim>
+    double
+    AdiabaticBoundary<dim>::initial_temperature (const Point<dim> &) const
+    {
+      Assert (false, ExcNotImplemented());
+      return 0;
+    }
+
+    /**
+     * Get the depth of the adiabatic boundary isotherm for the current lat/long.
+     * If above the isotherm use a linear interpolation to the surface.
+     * If below the isotherm use an increase of .5 degrees per kilometer.
+     */
+
+    template <>
+    double
+    AdiabaticBoundary<3>::initial_temperature (const Point<3> &position) const
+    {
+      const std::pair<double, double> lat_long = lat_long_from_xyz_WGS84(position);
+      const double depth                       = position.norm() - radius_WGS84(lat_long.first);
+      const double isotherm_depth              = get_isotherm_depth(lat_long.first, lat_long.second);
+      if (depth < isotherm_depth)
+        return isotherm_temperature - (depth - isotherm_depth) * 0.0005;
+      else
+        return surface_temperature + ( depth/isotherm_depth)*(isotherm_temperature - surface_temperature);
+    }
+
+
+    template <int dim>
+    void
+    AdiabaticBoundary<dim>::declare_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("Initial conditions");
+      {
+        prm.enter_subsection("Adiabatic boundary");
+        {
+          prm.declare_entry ("Isotherm depth filename",
+                             "adiabatic_boundary.txt",
+                             Patterns::FileName(),
+                             "File from which the base of the lithosphere depth data is read. "
+                             "Note that full path (/$DIRECTORY_PATH/filename) of the location "
+                             "of the file must be provided. ");
+          prm.declare_entry ("Isotherm temperature", "1673.15",
+                             Patterns::Double (0),
+                             "The value of the isothermal boundary temperature. Units: Kelvin. ");
```

Comment:
Please remove the space at the end of the string. It is only needed if you want to continue the string in the next line.